pkg/interfaces/contracts/pool-utils/IManagedPool.sol

```diff
@@ -221,6 +228,12 @@ interface IManagedPool is IBasePool {
      * @dev This can be called by anyone to collect accrued AUM fees - and will be called automatically
      * whenever the supply changes (e.g., joins and exits, add and remove token), and before the fee
      * percentage is changed by the manager, to prevent fees from being applied retroactively.
+     *
+     * Correct behavior depends on the current supply, which is potentially manipulable if the pool
+     * is reentered during execution of a Vault hook.
```

Comment:
I know this is an implementation detail and not necessarily part of the interface, but since we are protecting this one, shouldn't we clarify that given the protection it is safe to use?

If I read this comment without all the context, I wouldn't know whether it is safe to call or not, nor how to call it safely.